src/sinks/util/sink.rs

```diff
@@ -129,6 +133,13 @@ impl<T: Sink> Sink for StreamSink<T> {
     }
 }
 
+// === StreamSink2 ===
+
+#[async_trait]
+pub trait StreamSink2 {
+    async fn run(&mut self, input: BoxStream<'_, Result<Event, ()>>) -> Result<(), ()>;
```

Comment:
Why do we have `Result<Event, ()>` in this type, rather than bare `Event`? Is it meaningful to pass errors to sinks like this?

Reply:
@lukesteensen ask me about same in DM :smile: 
The problem in `futures01::Stream` which requires 2 types: `Item` and `Error`. These 2 types converted with compatibility code to `Result<Item, Error>`. I decide to leave `Result` for some time, but maybe I should delete it already and add extra code for compatibility with futures01.

Reply:
I'd suggest we map the `Err(())` of `futures01::Stream` to `None`, and unwrap the `Ok`. It's a universally applicable conversion, meaningful if we bound the error type to `()` (which we do).

I suppose the sooner we sweep this `Result` out the better. It will allow us to get rid of the odd (and potentially erroneous) logic at the sink level.

Reply:
Yeah, I think you are right. Sooner is better, will change :+1: